gslib/tests/test_creds_config.py

```diff
@@ -62,9 +118,58 @@ def testExactlyOneInvalid(self):
     with SetBotoConfigForTest(
         [('Credentials', 'gs_oauth2_refresh_token', 'foo'),
          ('Credentials', 'gs_service_client_id', None),
-         ('Credentials', 'gs_service_key_file', None)],
+         ('Credentials', 'gs_service_key_file', None),
+         ('Credentials', 'gs_impersonate_service_account', None)],
         use_existing_config=False):
       stderr = self.RunGsUtil(['ls', suri(bucket_uri)],
                               expected_status=1,
                               return_stderr=True)
       self.assertIn('credentials are invalid', stderr)
+
+  @unittest.skipUnless(SERVICE_ACCOUNT,
+                       'Test requires service account configuration.')
+  @SkipForS3('Tests only uses gs credentials.')
+  @SkipForXML('Tests only run on JSON API.')
+  def testImpersonationCredentialsFromBotoConfig(self):
+    with SetBotoConfigForTest([('Credentials', 'gs_impersonate_service_account',
+                                SERVICE_ACCOUNT)]):
+      with SetEnvironmentForTest({}):
+        stderr = self.RunGsUtil(['ls', 'gs://pub'], return_stderr=True)
+        self.assertIn('using service account impersonation', stderr)
```

Comment:
Do you think it's sufficient to check for this message?  It doesn't guarantee that the impersonation credentials are actually used.